packages/host/app/services/command-service.ts

```diff
@@ -0,0 +1,68 @@
+import Service from '@ember/service';
+
+type Command = 'patch';
+
+class CommandHandler {
+  handler: any;
+  functions: Map<Command, ((arg: any) => void)[]>;
+  constructor(handler: any) {
+    this.handler = handler;
+    this.functions = new Map();
+  }
+
+  public get(command: Command) {
+    if (!this.functions.has(command)) {
+      this.functions.set(command, []);
+    }
+    return this.functions.get(command);
+  }
+
+  public async handle(command: Command, arg: any) {
+    if (!this.functions.has(command)) {
+      throw new Error(`No handler registered for command ${command}`);
+    }
+    const functions = this.functions.get(command)!;
+    for (const func of functions) {
+      await func(arg);
+    }
+  }
+  public setHandler(command: Command, func: (arg: any) => void) {
+    if (!this.functions.has(command)) {
+      this.functions.set(command, []);
+    }
+    this.functions.get(command)!.push(func);
+  }
+}
+
+export default class CommandService extends Service {
+  commandHandlers: Map<any, CommandHandler>;
+
+  constructor(properties: object) {
+    super(properties);
+    this.commandHandlers = new Map();
+  }
+
+  public registerCommandHandler(
+    handler: any,
+    command: Command,
+    func: (arg: any) => void,
```

Comment:
hmmm, you are passing async functions into here. usually event handlers as synchronous and if you need to do async work then you have to esatblish an async drain (like i mentioned in the aibot work). You could make this an async handler, but that does mean that your subscribers will not all be executed in a synchronous fashion--async from one subscriber needs to be absorbed before handling another subscriber for the same event (which is generally not ideal).